mediapipe/graphs/instantmotiontracking/calculators/gl_animation_overlay_calculator.cc

```diff
@@ -294,7 +294,98 @@ bool GlAnimationOverlayCalculator::LoadAnimationAndroid(
       return false;
     }
 
+    // Use the vertices in order to generate normals for shader usage.
     triangle_mesh.normals.reset(new float[lengths[0]]);
+    float vertex_normals[lengths[0]];
+    float triangle_surface_normals[lengths[0]];
+    int len = lengths[0];
+
+    // Compute every triangle surface normal and store them for later use.
+    for (int i = 0; i < len; i += 3) {
+      // Vector A = V2 - V1
+      float Ax, Ay, Az;
+      // Vector B = V3 - V1;
+      float Bx, By, Bz;
+      // Calculate V1 (Previous Vertex), V2 (Current Vertex), V3 (Next Vertex)
+      // Last vertex (Edge case)
+      if(i >= len-3) {
```

Comment:
We don't need 3 copies of this code here.  We could either refactor the calculations into their own function, or better yet, we could just use the % operator to wrap around automatically.